src/backend/access/transam/xlog.c

```diff
@@ -181,6 +183,16 @@ const struct config_enum_entry recovery_target_action_options[] = {
 	{NULL, 0, false}
 };
 
+
+typedef struct LastWrittenLsnCacheEntry
+{
+	BufferTag	key;
```

Comment:
Hmm, the block number in the BufferTag is the first block of the 8 MB chunk of blocks, right? Would be good to add a comment here. Or maybe at the top of the struct, to explain how the cache works in general.